### PR TITLE
Task properties: use GtkComboBox instead of GtkOptionMenu

### PR DESCRIPTION
**Pull request by oskarb**
GtkOptionMenu has been deprecated since forever. This replaces it with GtkComboBox in props-task.c.

The existing code attached the numeric values to the menu items. In the *save* code path, it grabbed the numeric value from the current selection. On the other hand, in the *load* code path, it assumed that the menu items appeared in a certain order.

The new code uses a GtkListStore with two columns, of which only one is visible. The first column is the display text, while the second column is the numeric value that will be stored in the task.

Various helper methods are introduced to work with this style of GtkListStore.

### CODE REVIEW

**Comment by oskarb** [comment]
@markuspg This explores a possible way to replace (some?) GtkOptionMenu with GtkComboBox (issue #12).

In task_properties.glade/props-task.c, each item essentially contained consisted of a display name and a numerical constant that used stored for the task. GtkComboBox works with a GtkListStore, and while it does offer a convenience specialization to easily get a combo box with just text items, I have not found a convenient way to work with this sort of tagged values, except for directly dealing with the GtkListStore complications.

This PR therefore explores creating a number of helper methods to work with comboboxes where the list store contains one display string and one hidden numerical value.

I continue to wonder if it really must be this hard? Is there something in the APIs that I have missed?

As alternative approaches, one could of course skip the numerical values, use a text-only combo box, and work with the item index. Or have a lookup table to search for the string in order to find the numerical constant. But the former feels like a hack, and the latter equally complex.

Obviously, the various helper methods in this branch needs cleanup for naming, error handling etc, and should be in a different file for reusability. But after that is done, perhaps this approach will work fairly well, with minimal amount of code at the point-of-use.

What do you think of this?

If you agree with this approach, I will clean it up.

**Comment by oskarb** [comment]
@markuspg I've cleaned it up a bit and moved the generic code the a new file. For sure there are probably things in select-list.c that could be cleaner. Though I think the use of it looks quite convenient in props-task.c and this should work for props-proj.c too. 

Still don't know if there is some other canonical way to doing it. Gtk3 has a built in combo box subclass that allows a display string and an id string, which looks like the same basic idea.

File: src/select-list.h
Comment (markuspg) [inline src/select-list.h]: I would advocate for just going with `GTT_SELECT_LIST_H`. I see that Gtk is doing it the way you did, but isn't a preceding ` __`  reserved by the language?

File: src/Makefile.am
Comment (markuspg) [inline src/Makefile.am]: Please add the `select-list.c` to the `CMakeLists.txt` file in this directory too.

File: src/select-list.h
Comment (markuspg) [inline src/select-list.h]: To me it would feel slightly cleaner to add this (and the corresponding source file) as `gtt-select-list.h`. But feel free to leave it as it is.

File: src/select-list.h
Comment (markuspg) [inline src/select-list.h]: This has the taste of creating a new type. I wonder if it would be cleaner to go this road entirely. It's just a thought, so feel welcome to leave it as it is.

When thinking about this issue I had the slight inclination to go even further. It seems that the `GtkOptionMenu`  instances are used at least partially to choose enumerators. So for maximum convenience for the trade off of some slight code bloat (here I miss generics in C ;-)) why not create specialized combo box types for the enumerations in question?

File: glade/task_properties.glade
Comment (markuspg) [inline glade/task_properties.glade:354]: Why this change? Doesn't the `GtkComboBox`  have useful actions on focus?

File: src/select-list.c
Comment (markuspg) [inline src/select-list.c]: Please format this with `clang-format`. There are only minor findings, but still it would be great to have this in `dev`  clean.

**Comment by markuspg** [comment]
To me your approach looks fine. The only difference in approach I would have (maybe) taken is creating dedicated combo box derivatives for the enumerations, but that's just an idea too.

Feel free to merge as is or rework as you want to. If you push updated code I'll strive to re-approve quickly.

I'm just starting a new job, so please apologize if my reviews are slightly delayed currently.

**Review by markuspg — DISMISSED** [review]

File: src/select-list.h
Comment (oskarb) [reply src/select-list.h]: Actually, I just copied from some other gnotime header file and when I check now, it seems most of gnotime's header files use this naming pattern. A quick web search finds various coding style definitions that vary on the subject. But yeah, it's supposed to be reserved.

File: src/select-list.h
Comment (oskarb) [reply src/select-list.h]: That makes sense.

File: src/select-list.h
Comment (oskarb) [reply src/select-list.h]: This is all deprecated again in Gtk 4.10, and the idea in Gtk4+ seems to be to use GtkDropDown and GListModel instead of GtkComboBox and GtkListStore. So for now I aimed for a solution that didn't duplicate too much code and that didn't make things worse (i.e. I wanted to avoid putting just strings in the list and rely on string matching or element order), and then in Gtk 4 there may be a different approach.

I miss generics too. :) Creating dedicated subclasses for the GtkComboBox may be overkill, especially since they are each used in just one place, but I was considering that a specialization to display any enum might be useful. Possibly based on GEnumValue. To be honest this seems like such a common concept that it's strange there is no existing convenience helper for it.

Comment (oskarb) [reply src/select-list.h]: So in summary, I think further improvements on this should wait until they can be done based on Gtk4.

File: glade/task_properties.glade
Comment (oskarb) [reply glade/task_properties.glade:354]: Keyboard navigation works correctly with can_focus false. Set it to true, and you cannot tab away from it. Nobody really knows why and the documentation appears useless on the subject but it's been noted (https://gitlab.gnome.org/GNOME/gtk/-/issues/2077).

File: src/select-list.c
Comment (oskarb) [reply src/select-list.c]: Done

**Comment by oskarb** [comment]
@markuspg I've done various small adjustments following your review, so it needs another review before I can merge it.

File: src/select-list.h
Comment (markuspg) [reply src/select-list.h]: Sounds fine :+1:

**Review by markuspg — APPROVED** [review]

**Pull request merged**
